mteb/evaluation/MTEB.py

```diff
@@ -470,35 +470,46 @@ def run(
 
             existing_results = None
             save_path = None
+            final_splits_to_run = task_eval_splits
+            missing_evaluations = self._get_missing_evaluations(
+                existing_results,
+                task_eval_splits,
+                task_subsets,
+                eval_subsets,
+            )
 
             if output_path:
                 save_path = output_path / f"{task.metadata.name}{task.save_suffix}.json"
                 if save_path.exists():
                     existing_results = TaskResult.from_disk(save_path)
 
-                    if not overwrite_results:
+                    # Unified call to get missing splits and subsets
+                    missing_evaluations = self._get_missing_evaluations(
+                        existing_results,
+                        task_eval_splits,
+                        task_subsets,
+                        eval_subsets,
+                    )
+
+                    # Determine final splits to run
+                    final_splits_to_run = []
+                    # We need to run any split that is fully missing or has missing subsets
+                    for sp, info in missing_evaluations.items():
+                        if info["whole_split_missing"] or info["missing_subsets"]:
+                            final_splits_to_run.append(sp)
+
+                    if overwrite_results:
+                        final_splits_to_run = task_eval_splits
```

Comment:
`final_splits_to_run = task_eval_splits` is already called on line 473. Maybe only run this part (L494-L499) when `if not overwrite_results`?